doc/sphinx/architecture.md

```diff
@@ -0,0 +1,252 @@
+- Directories:
+  - `include/server`
+  - `src/server`
+  - `src/include/server`
+  
+## mircompositor
+- Included in `mirserver.so`
+- The `Compositor` is responsible for collecting the set of renderables and sending them off to the outputs to be displayed.
+- Directories:
+  - `src/include/server/mir/compositor`
+  - `src/server/compositor`
+
+## mirinput
+- Included in `mirserver.so`
+- Deals with everything input (e.g. keyboard presses, mouse clicks)
+- The `InputManager` provides access to the specific input platform that is running (e.g. X, Wayland, or libinput).
+- The `InputDispatcher` is responsible for taking events bubbled up from the platform and sending them off to the appropriate listeners
+- Directories
+  - `src/include/server/mir/input`
+  - `src/server/input`
+  
+## mirfrontend
+- TODO: How do I describe this one?
+- Directories:
+  - `src/include/server/mir/frontend`
+  - `src/server/frontend`
+  
+## mirfrontend-wayland
+- The *Mir* implementation of the wayland protocol.
+- The `WaylandConnector` class is the glue between the compositor and the specific details of the wayland protocol.
+- Directories:
+  - `src/server/frontend_wayland`
+
+## mirfrontend-xwayland
+- The *Mir* implementation of the xwayland protocol.
+- The `XWaylandConnector` class is the glue between the compositor and the specific details of the xwayland protocol.
+- This protocol talks to the `xwayland` server, which is spawned as a subprocess.
+- Directories:
+  - `src/server/frontend_xwayland`
+
+## mirshell
+- The `Shell` is responsible for managing the surfaces that the compositor wants to show.
+- It provides an interface to create, update, and remove these surfaces from the display.
+- As an example, the `WaylandConnector` might ask the `Shell` to create a new surface for it.
+- Directories:
+  - `src/include/server/mir/shell`
+  - `src/server/shell`
+  
+## mirshelldecoration
+- Manages the decoration of *Mir*'s windows
+- Directories:
+  - `src/server/shell/decoration`
+
+## mirscene
+- The `Scene` provides an interface for the `Compositor` to access the list of renderable items
+  - These renderables are derived from the surfaces that were added to the `Shell`.
+  - You can think of the `Scene` as you would think of a "scene graph" in a 3D game engine.
+- Unlike the `Shell`, the `Scene` knows a lot about the Z-order of the surfaces. For this reason, it is also responsible for things like locking the display or sending surfaces to the back of the Z-order.
+- Directories:
+  - `src/include/mir/server/scene`
+  - `src/server/scene`
+
+## mirgraphics
+- A graphics abstraction layer sitting between the compositor and the specific graphics platform that it is using to render.
+- Directories:
+  - `src/include/mir/server/graphics`
+  - `src/server/graphics`
+  
+## mirreport
+- Default logging and reporting facilities
+- Directories:
+  - `src/server/report`
+
+## mirlttng
```

Comment:
I like this naming even less. `lttng` is a directory/namespace within `report` - it shouldn't be named at the same level as `mirwayland`. Similarly, `logging`, `null`.